### DISCUSSION TITLE
Changing item meta

### DISCUSSION THREAD

**Discussion by TogAr2** [Minestom/Minestom#922] [Help] [answered]
With the new item meta api, is it possible to simplify this?
```java
public static ItemStack setCrossbowCharged(ItemStack stack, boolean charged) {
	return stack.withMeta(stack.meta().with(meta -> new CrossbowMeta.Builder(meta.tagHandler()).charged(charged).build()));
}
```

Because it is much longer than the previous equivalent and it uses 2 experimental methods.

**Comment by TheMode** [comment, marked as answer]
`ItemMeta` is a sealed interface, meaning that the object retrieved from `ItemStack#meta` cannot be casted (anymore)

The specialized meta classes are still available in the `item.metadata` package, but are only considered as view (i.e. interact with the item NBT, without the item actively depending on it). `ItemStack#withMeta` give you the ability to provide a `Class<ItemMetaView>` to access such specialized view
```java
public static ItemStack setCrossbowCharged(ItemStack item, boolean charged) {
    return item.withMeta(CrossbowMeta.class, builder -> builder.charged(charged));
}
```
And `ItemStack#meta(Class)` if you do not need to modify the meta
```java
CrossbowMeta crossbowMeta = item.meta(CrossbowMeta.class);
```

The change also allow you to make your own `ItemMetaView` (see the existing classes such as `CrossbowMeta`) without having to mess with the registry or minestom internal.
